### PR TITLE
Add tweetnacl/20140427 recipe

### PR DESCRIPTION
**Pull request by madebr**
Specify library name and version:  **tweetnacl/20140427**

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/wiki#how-to-submit-a-pull-request) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/tweetnacl/all/conanfile.py [hunk 1]
```diff
@@ -0,0 +1,68 @@
+from conans import ConanFile, CMake, tools
+import os
+
+
+TWEETNACL_LICENSE = """
```
Comment (uilianries) [inline recipes/tweetnacl/all/conanfile.py]: Remove it and download the license from https://unlicense.org/UNLICENSE

File: recipes/tweetnacl/all/conanfile.py [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+    version = "20140427"
+    license = "Public Domain"
+    homepage = "https://tweetnacl.cr.yp.to"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "TweetNaCl is the world's first auditable high-security cryptographic library"
+    exports = ["PUBLIC_DOMAIN_LICENSE.md", "LICENSE.md"]
+    exports_sources = ["CMakeLists.txt"]
+    generators = "cmake"
+    options = {
+        "fPIC": [True, False],
```
Comment (uilianries) [inline recipes/tweetnacl/all/conanfile.py]: Why not shared?

File: recipes/tweetnacl/all/conanfile.py [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+
+For more information, please refer to http://unlicense.org/
+"""
+
+
+class TweetnaclConan(ConanFile):
+    name = "tweetnacl"
+    version = "20140427"
```
Comment (uilianries) [inline recipes/tweetnacl/all/conanfile.py]:
```suggestion
```

No version should be declared when using `all` folder

File: recipes/tweetnacl/all/conanfile.py [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+    license = "Public Domain"
+    homepage = "https://tweetnacl.cr.yp.to"
+    url = "https://github.com/conan-io/conan-center-index"
+    description = "TweetNaCl is the world's first auditable high-security cryptographic library"
+    exports = ["PUBLIC_DOMAIN_LICENSE.md", "LICENSE.md"]
```
Comment (uilianries) [inline recipes/tweetnacl/all/conanfile.py]:
```suggestion
```

File: recipes/tweetnacl/all/test_package/test_package.cpp [hunk 5]
```diff
@@ -0,0 +1,105 @@
+#ifdef __cplusplus
```
Comment (uilianries) [inline recipes/tweetnacl/all/test_package/test_package.cpp]: You can change it to a C file instead, simpler and fits as well.

File: recipes/tweetnacl/all/conanfile.py (hunk 2 above)
Comment (madebr) [reply recipes/tweetnacl/all/conanfile.py]: It won't work on Visual Studio due to missing __declspec(dllimport).

Comment (uilianries) [reply recipes/tweetnacl/all/conanfile.py]: So exclude it only for Visual Studio ...

File: recipes/tweetnacl/all/conanfile.py [hunk 6]
```diff
@@ -0,0 +1,52 @@
...
+    settings = "os", "compiler", "build_type", "arch"
+
+    _cmake = None
+
+    def config_options(self):
+        if self.settings.os == "Windows":
+            del self.options.fPIC
+
+    def configure(self):
+        del self.settings.compiler.libcxx
```
Comment (uilianries) [inline recipes/tweetnacl/all/conanfile.py]:
```suggestion
        del self.settings.compiler.libcxx
        del self.settings.compiler.cppstd
```

File: recipes/tweetnacl/all/conanfile.py (hunk 2 above)
Comment (madebr) [reply recipes/tweetnacl/all/conanfile.py]: It is excluded for Windows because it needs a randombytes implementation.

**Comment by madebr** [comment]
@uilianries I think the buid error is due to the manual removal of the `shared` options on Windows.

**Comment by uilianries** [comment]
> @uilianries I think the build error is due to the manual removal of the `shared` option on Windows.

raise a ConanInvalidConfiguration rather than removing `shared` on Windows.

**Comment by Croydon** [comment]
https://github.com/conan-community/conan-tweetnacl should be archived now

**Comment by uilianries** [comment]
> [conan-community/conan-tweetnacl](https://github.com/conan-community/conan-tweetnacl) should be archived now

Done!

**Pull request merged**